src/journal/journald-socket.c

```diff
@@ -13,39 +13,38 @@
 #include "socket-util.h"
 #include "sparse-endian.h"
 
-void server_open_forward_socket(Server *s) {
+static int server_open_forward_socket(Server *s) {
         _cleanup_close_ int socket_fd = -EBADF;
         const SocketAddress *addr;
         int family;
 
         assert(s);
 
-        /* nop if there is nothing to do */
-        if (s->forward_to_socket.sockaddr.sa.sa_family == AF_UNSPEC || s->namespace || s->forward_socket_fd >= 0)
-                return;
+        /* Noop if there is nothing to do. */
+        if (s->forward_to_socket.sockaddr.sa.sa_family == AF_UNSPEC || s->namespace)
+                return 0;
+        /* All ready, nothing to do. */
+        if (s->forward_socket_fd >= 0)
+                return 1;
 
         addr = &s->forward_to_socket;
 
         family = socket_address_family(addr);
 
-        if (!IN_SET(family, AF_UNIX, AF_INET, AF_INET6, AF_VSOCK)) {
-                log_debug("Unsupported socket type for forward socket: %d", family);
-                return;
-        }
+        if (!IN_SET(family, AF_UNIX, AF_INET, AF_INET6, AF_VSOCK))
+                return log_debug_errno(SYNTHETIC_ERRNO(ESOCKTNOSUPPORT),
+                                       "Unsupported socket type for forward socket: %d", family);
 
         socket_fd = socket(family, SOCK_STREAM|SOCK_CLOEXEC, 0);
-        if (socket_fd < 0) {
-                log_debug_errno(errno, "Failed to create forward socket, ignoring: %m");
-                return;
-        }
+        if (socket_fd < 0)
+                return log_debug_errno(errno, "Failed to create forward socket, ignoring: %m");
 
-        if (connect(socket_fd, &addr->sockaddr.sa, addr->size) < 0) {
-                log_debug_errno(errno, "Failed to connect to remote address for forwarding, ignoring: %m");
-                return;
-        }
+        if (connect(socket_fd, &addr->sockaddr.sa, addr->size) < 0)
+                return log_debug_errno(errno, "Failed to connect to remote address for forwarding, ignoring: %m");
 
         s->forward_socket_fd = TAKE_FD(socket_fd);
-        log_debug("Successfully connected to remote address for forwarding");
+        log_debug("Successfully connected to remote address for forwarding.");
+        return 1;
 }
 
 static inline bool must_serialise(struct iovec iov) {
```

Comment:
While at it, maybe -> `serialize` for consistency with the rest of the codebase?